src/test_utils.rs

```diff
@@ -2,17 +2,21 @@
 // so that tests can be run with all the necessary resources.
 // This removes the requirement for having to limit test threads
 // or to build/run nym-client ourselves.
-#[macro_export]
-macro_rules! new_nym_client {
-    ($nym_id:ident, $uri:ident) => {
-        let docker_client = clients::Cli::default();
-        let nym_ready_message = WaitFor::message_on_stderr("Client startup finished!");
-        let nym_image = GenericImage::new("nym", "latest")
-            .with_env_var("NYM_ID", $nym_id)
-            .with_wait_for(nym_ready_message)
-            .with_exposed_port(1977);
-        let nym_container = docker_client.run(nym_image);
-        let nym_port = nym_container.get_host_port_ipv4(1977);
-        let $uri = format!("ws://0.0.0.0:{nym_port}");
-    };
+
+use testcontainers::{clients::Cli, core::WaitFor, images::generic::GenericImage, Container};
+
+/// Create a nym client using the same docker Cli
+pub fn create_nym_client<'a>(
+    docker_client: &'a Cli,
```

Comment:
Yes, this is necessary, and it has better ergonomics than creating a local variable tagged with `#[allow(unused)]`.

The `Cli` and the `Container` will need to have the same lifetime since the `Container` have a reference to the docker `Cli`.

Sure, we can create the `Cli` in the `create_nym_client`, but we will not be able to return `Container` object out of this function since the `Cli` inside of `create_nym_client` would have already been dropped after the function call.

On the other hand, by not returning the `Container` object will cause it be dropped before the return of the function. Any further interaction with the container will fail, since the `Container` drop is implemented to remove it's corresponding running docker container.

So, the logical code would be to instantiate a docker `Cli` in where the `create_nym_client` function is called. Although the container is created inside of `create_nym_client` the container is returned at the end of the function in which the ownership of `Container` is now the caller function, which owns both the `Cli` and also the `Container.`